src/utils/url-utils.ts

```diff
@@ -68,6 +68,64 @@ function getFilenameIndexInUrl(url: string): number {
   return indexOfLastSlash + 1;
 }
 
+/**
+ * Take two URLs and try to construct a relative URL for the second (`newUrl`)
+ * relative to the first (`baseUrl`).
+ *
+ * Returns `null` if they appear to be on different domains, depend on
+ * different schemes or if we don't have enough information to compute the
+ * relative URL.
+ * @param {string} baseUrl
+ * @param {string} newUrl
+ * @returns {string}
+ */
+function getRelativeUrl(baseUrl: string, newUrl: string): string | null {
+  const baseParts = parseURL(baseUrl);
+  const newParts = parseURL(newUrl);
+  if (
+    baseParts.scheme !== newParts.scheme ||
+    baseParts.authority !== newParts.authority
+  ) {
+    return null;
+  }
+
+  if (
+    (baseParts.path[0] === "/" || newParts.path[0] === "/") &&
+    baseParts.path[0] !== newParts.path[0]
+  ) {
+    return null;
+  }
```

Comment:
I'll add tests to check

Reply:
What I just pushed seems to fix the issue, though it's hard to wrap my head against all those cases

Reply:
It "fixes" this particular test case but I'm not sure it's the expected fix, the authority length is always >0 because its the part "github.com" here.

Reply:
getRelativeUrl(
        "http://github.com/"
        "http://github.com/rx-player",
      )
this gives "../rx-player" it does not seems right

Reply:
> the authority length is always >0 because its the part "github.com" here

Yes that was on purpose, because I wanted to return `null` if for example `getRelativeUrl("a/b", "/a/c")` is provided.

> [...] this gives "../rx-player" it does not seems right

You're right, are you succeeding to see those cases just by reading the code here?
I'm unable to, if it's easier for you to understand the logic I could let you write it.

I added the breaking test